src/devices/came_top432.c

```diff
@@ -0,0 +1,169 @@
+(describe the modulation, timing, and transmission, e.g.)
+The device uses PPM encoding,
+- 0 is encoded as 320 us gap and 640 us pulse,
+- 1 is encoded as 640 us gap and 320 us pulsep.
+The device sends a 4 times the packet when a button on the remote control is pressed.
+A transmission starts with a 320 us pulse. At the end of the packet, there is 36 periods of 320us between messages (11520us)
+
+Diagram :
+          |        <---> logical 0          <---> logical 1                     <-- 11520us -->
+          |       _   __    _   __   __   __    _    _    _   __   __    _    _                 _
+Pulse/gap | _____| |_|  |__| |_|  |_|  |_|  |__| |__| |__| |_|  |_|  |__| |__| |_____(..)______| |
+          +------------------------------------------------------------------------------   
+bits              ~~ : start bit (320us)
+                    |  0| 1  |  0 |  0 |  0 |  1 |  1 |  1 |  0 |  0 |  1 |  1 |
+                    |        0x4       |       0x7         |       0x3         |
+
+Data layout:
+    ccc
+- c : 4-bit data
+
+There is no CRC, no parity, no preamble (only the start pulse indicate the begining of packet)
+*/
+#include "decoder.h"
+
+#define CAMETOP432_BITLEN      12
+
+static int came_top432_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    data_t *data;
+    int r; // a row index
+    uint8_t *b; // bits of a row
+    int parity;
+    uint8_t r_crc, c_crc;
+    uint16_t sensor_id;
+    uint8_t msg_type;
+    int16_t code;
+    char code_str[3];
+    
+    if (decoder->verbose > 1) {
+        bitbuffer_printf(bitbuffer, "%s: ", __func__);
+    }
+
+    for (r = 0; r < bitbuffer->num_rows; ++r) {
+        b = bitbuffer->bb[r];
+
+        /*
+         * Validate message and reject invalid messages as
+         * early as possible before attempting to parse data.
+         *
+         * Check "message envelope"
+         * - valid message length (use a minimum length to account
+         *   for stray bits appended or prepended by the demod)
+         * - valid preamble/device type/fixed bits if any
+         * - Data integrity checks (CRC/Checksum/Parity)
+         */
+
+        if (bitbuffer->bits_per_row[r] < CAMETOP432_BITLEN) {
+            if (decoder->verbose > 1) {
+                fprintf(stderr, "%s: bitbuffer len received %d, expected : %d\n", __func__, bitbuffer->bits_per_row[r], CAMETOP432_BITLEN);
+            }
+            return DECODE_ABORT_EARLY;
+        }
+        // no preamble
+        // no crc
+        // no checksum
+        // no parity
+    }
+
+    // packet is not repeated
+    b = bitbuffer->bb[0];
+
+    // reconstruct the 12 bits code
+    code = ((b[0] << 4) | (b[1]>>4)) & 0xFFF;
+    sprintf(code_str, "%03x", code);
+    /* clang-format off */
+    data = data_make(
+            "brand",    "", DATA_STRING,   "CAME",
+            "model",    "", DATA_STRING,   "TOP432",
+            "type",     "", DATA_STRING,   "remote control",
```

Comment:
I would remove this, but I'm not fully sure. @zuckschwerdt we don't typically not the device type here, right? I think this field is meant for the sub-type of the model, because that's applicable in some cases.

Reply:
model should be "CAME-TOP432, no brand, no type.